### PR TITLE
feat: add Virtualizer component

### PR DESCRIPTION
**Pull request by ValeraS**

### CODE REVIEW

File: src/components/lab/Virtualizer/useLoadMore.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                onLoadMore();
+            }
+        };
+        element.addEventListener('scroll', onScroll);
+        return () => {
+            element.removeEventListener('scroll', onScroll);
+        };
+    }, [scrollContainerRef, onLoadMore, scrollOffset]);
+
+    const loadingRef = React.useRef(loading);
```
Comment (amje) [inline src/components/lab/Virtualizer/useLoadMore.tsx:52]: Why we need two similar refs?

Comment (ValeraS) [reply src/components/lab/Virtualizer/useLoadMore.tsx:52]: loadingRef contains the previous value of the loading prop to detect its changes, isLoadingRef is the current loading state.

File: src/components/lab/Virtualizer/Virtualizer.tsx [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+    scrollToIndex: (index: number, align?: ScrollAlignment) => void;
+    scrollOffset: number | null;
+    scrollRect: Rect | null;
+}
+
+interface VirtualizerProps extends Loadable, React.HTMLAttributes<HTMLDivElement> {
+    /** The ref of the virtualizer api. */
+    apiRef?: React.Ref<VirtualizerApi>;
+    /** The ref of the scroll container element. */
+    containerRef?: React.Ref<HTMLElement>;
```
[the remark this answers is not in the document]
Comment (ValeraS) [reply src/components/lab/Virtualizer/Virtualizer.tsx:34]: I don't see any benefit from this.

File: src/components/lab/Virtualizer/useLoadMore.tsx (hunk 1 above)
Comment (amje) [reply src/components/lab/Virtualizer/useLoadMore.tsx:52]: Maybe name it `prevLoadingRef` or `propLoadingRef`?

**Review by amje — APPROVED** [review]

**Pull request merged**